pkg/controller/tsp/tsp_controller.go

```diff
@@ -102,23 +77,14 @@ func (tc *Controller) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 // sync the config to predictor
-func (tc *Controller) syncTimeSeriesPrediction(prediction *predictionv1alph1.TimeSeriesPrediction) {
+func (tc *Controller) syncTimeSeriesPrediction(ctx context.Context, prediction *predictionv1alph1.TimeSeriesPrediction) (ctrl.Result, error) {
 	key := GetTimeSeriesPredictionKey(prediction)
 
 	c := NewMetricContext(prediction)
 
 	last, ok := tc.tsPredictionMap.Load(key)
 	if !ok { // first time created or system start
-
 		c.WithApiConfigs(prediction.Spec.PredictionMetrics)
-		//newStatus := prediction.Status.DeepCopy()
-		//cond := &predictionv1alph1.TimeSeriesPredictionCondition{
-		//	Type:               predictionv1alph1.TimeSeriesPredictionConditionNotReady,
-		//	Status:             metav1.ConditionTrue,
-		//	LastProbeTime:      metav1.Now(),
-		//	LastTransitionTime: metav1.Now(),
-		//}
-		//UpdateTimeSeriesPredictionCondition(newStatus, cond)
 	} else {
 		if old, ok := last.(*predictionv1alph1.TimeSeriesPrediction); ok {
```

Comment:
which scene need to compare old prediction and new prediction here?

Reply:
delete the old configs, because the internal config is cached in memory in prediction